canvas/models/models.py

```diff
@@ -148,6 +148,11 @@ def total_tokens_received(self):
         for uqj in practiced_uqjs:
             tokens += uqj.tokens_received
 
+        event_sets = EventSet.objects.filter(course=self.course).all()
```

Comment:
I think here you are double-counting some tokens. If an event is in an event set you count it both individually and from the event.